### DISCUSSION TITLE
Read from XML file #26

### DISCUSSION THREAD

**Discussion by fk250046** [wixtoolset/issues#7222] [WiX Development]
I would like to finish this development I sugested several years ago and I want to discuss architecture here because XML schema should be changed. Since I started development of this I think that what I suggested in first document should be changed as following:
1. The table can conatin 5 columns:
    a) Property - Identificator, primary key, no validation against Property table, results set to it.
    b) File - Formatted string of [MAX_PATH], Path to XML file, cannot be nullable
    c) Element - String, No limitation in length, XPath to an XML element, cannot be nullable.
    d) Name - String, No limitation in length, name of attribute to read, nullable, if
         i. Name=<value> - reads a value from attribute <value>
         ii. Name=null - reads inner text from element
         iii. Name="*" - reads all child nodes
     e) Hidden - yes/no type, adds this property to hidden properties.
2. Custom action - immediate, after AppSearch in both sequences, run once, check, does not fail installation if finds nothing but fails installation if there is a problem finding/using XML DOM.

**Comment by barnson** [comment]
What's `#26`?

**Reply by fk250046** [reply]
This was the number of new feature issue in Wix 3, I added it here as a reference

**Reply by barnson** [reply]
Sorry, not sure what you're referring to. Do you have a link?

**Comment by fyodorkor** [comment]
Sorry I mislead you, here is the link for feature request
https://github.com/wixtoolset/issues/issues/5494

On Thu, Feb 16, 2023 at 7:25 PM Bob Arnson ***@***.***> wrote:

> Sorry, not sure what you're referring to. Do you have a link?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/wixtoolset/discussions/7222#discussioncomment-4998949>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACCKQ2YZGD7J67XFZPM6MA3WXZPH5ANCNFSM6AAAAAAU52GBXU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Regards,
Fyodor Koryazhkin..

**Reply by barnson** [reply]
Got it, thanks! I'll mark it for triage next week.
